src/pushsource/_impl/backend/koji_source.py

```diff
@@ -509,6 +521,94 @@ def _push_items_from_container_build(self, nvr, meta):
 
         return out
 
+    def _push_items_from_vmi_build(self, nvr, meta):
+        LOG.debug("Looking for virtual machine images on %s, %s", nvr, meta)
+
+        if not meta:
+            message = "Virtual machine image build not found in koji: %s" % nvr
+            LOG.debug(message)
+            raise ValueError(message)
+
+        # VMI has the {'typeinfo': {'image':{}}} on extra
+        extra = meta.get("extra") or {}
+        typeinfo = extra.get("typeinfo") or {}
+        image = typeinfo.get("image")
+
+        if image is None:
+            message = "Build %s not recognized as a virtual machine image build" % nvr
+            LOG.debug(message)
+            raise ValueError(message)
```

Comment:
It would be nice to filter out container builds here as well.
* `meta['container_koji_task_id']` is available just for container builds
* https://github.com/containerbuildsystem/atomic-reactor/blob/master/docs/koji.md?plain=1#L79

I know ideal would be if could images has deterministic filed in extra but one cannot rely on `{'typeinfo': {'image':{}}}` yet.

It might simplify additional logic for archives.

Reply:
Oh, thank you very much for letting me know this!

I was trying to filter out by the extension (tar) which isn't a great way to do it, so I'll adjust